### PR TITLE
feat: add trace to logs

### PR DESCRIPTION
**Pull request by shuli-ogp**
## Problem

Implements tracing for (https://github.com/datagovsg/formsg-private/issues/25)

## Solution

Add trace to logs using cloudflare cf-ray header. If cf-ray does not exist, trace using x-request-id header

## Tests

**On staging**
- [ ] At the login page, key in an invalid email address abc@abc.com. Check that the value of cf-ray header is logged under trace in the server logs error message.
- [ ] Access a public form. Check that the value of cf-ray header is logged under trace in the server logs for the get request.

**On localhost**
- [ ] At the login page, key in an invalid email address abc@abc.com. Check that the value of x-request-id header is logged under trace in the server logs error message.
- [ ] In L162-L164 of `logger.ts`, replace
```
isDev
  ? format.combine(format.colorize(), errorPrinter(), customFormat)
  : format.json({ replacer: jsonErrorReplacer }),
```
with 
``` 
format.json({ replacer: jsonErrorReplacer }),
```
Access a public form. Check that the value of x-request-id header is logged under trace in the server logs for the get request. 

## New dependencies

- express-request-id^1.4.1

### CODE REVIEW

File: src/app/utils/request.ts [hunk 1]
```diff
@@ -1,5 +1,14 @@
 import { Request } from 'express'
 
-export const getRequestIp = (req: Request) => {
+interface IRequestWithId extends Request {
```
Comment (karrui) [inline src/app/utils/request.ts:3]: nit: Can we extend the namespace globally (following suggestions in this [SO post](https://stackoverflow.com/questions/37377731/extend-express-request-object-using-typescript), some may be outdated) so it's consistent throughout the application?

Can also be done in a different PR, since I think we need to extend request to also have `req.session.user`

File: src/app/controllers/admin-console.server.controller.js [hunk 2]
```diff
@@ -577,6 +577,7 @@ exports.getExampleFormsUsingAggregateCollection = function (req, res) {
           meta: {
             action: 'getExampleFormsUsingAggregateCollection',
             ip: getRequestIp(req),
+            trace: getTrace(req),
             url: req.url,
```
Comment (karrui) [inline src/app/controllers/admin-console.server.controller.js:581]: Out of scope for this PR, but can you help create an issue for an utility function that returns the relevant log info from a `req`? Seems like the ip, trace, url, header combo is repeated throughout the application :P

**Review by karrui — APPROVED** [review]
lgtm

File: src/app/utils/request.ts (hunk 1 above)
Comment (shuli-ogp) [reply src/app/utils/request.ts:3]: Ok! Will create issue for this and work on ti

File: src/app/controllers/admin-console.server.controller.js (hunk 2 above)
Comment (shuli-ogp) [reply src/app/controllers/admin-console.server.controller.js:581]: Ok!
